fullstory-segment-middleware/src/test/java/com/fullstorydev/fullstory_segment_middleware/FullStorySegmentMiddlewareTest.java

```diff
@@ -0,0 +1,443 @@
+        // see Mokito document here: https://github.com/powermock/powermock/wiki/mockito#a-full-example-for-mocking-stubbing--verifying-static-method
+        FS.setUserVars(userVars);
+    }
+
+    @Test
+    public void intercept_GroupPayloadChain_EnableGroupTraitsAsUserVars_ChainProceedCalled() {
+        fullStorySegmentMiddleware = new FullStorySegmentMiddleware(mockContext, "SegmentWriteMockKey");
+        fullStorySegmentMiddleware.enableGroupTraitsAsUserVars = true;
+        fullStorySegmentMiddleware.enableFSSessionURLInEvents = false;
+        Map<String, String> map = new HashMap<>();
+        map.put("industry", "retail");
+        GroupPayload groupPayload = new GroupPayload.Builder()
+                .userId("userId")
+                .groupId("groupId")
+                .traits(map)
+                .build();
+        when(mockChain.payload()).thenReturn(groupPayload);
+        fullStorySegmentMiddleware.intercept(mockChain);
+        // verify that we did not block chain.proceed, nor modified the payload
+        verify(mockChain, times(1)).proceed(groupPayload);
+    }
+
+    @Test
+    public void intercept_GroupPayloadChain_EnableGroupTraitsAsUserVars_FSSetUserVarsCalled() {
+        fullStorySegmentMiddleware = new FullStorySegmentMiddleware(mockContext, "SegmentWriteMockKey");
+        fullStorySegmentMiddleware.enableGroupTraitsAsUserVars = true;
+        fullStorySegmentMiddleware.enableFSSessionURLInEvents = false;
+        Map<String, String> map = new HashMap<>();
+        map.put("industry", "retail");
+        GroupPayload groupPayload = new GroupPayload.Builder()
+                .userId("userId")
+                .groupId("groupId")
+                .traits(map)
+                .build();
+        when(mockChain.payload()).thenReturn(groupPayload);
+        fullStorySegmentMiddleware.intercept(mockChain);
+
+        Map<String, String> suffixedMap = new HashMap<>();
+        suffixedMap.put("groupID_str", groupPayload.groupId());
+        suffixedMap.put("industry_str", "retail");
+
+        verifyStatic(FS.class, VerificationModeFactory.times(1));
+        // IMPORTANT:  Call the static method you want to verify.
+        // see Mokito document here: https://github.com/powermock/powermock/wiki/mockito#a-full-example-for-mocking-stubbing--verifying-static-method
+        FS.setUserVars(suffixedMap);
+    }
+
+    @Test
+    public void intercept_IdentifyPayloadChain_NoTraits_FSIdentifyCalled() {
+        fullStorySegmentMiddleware = new FullStorySegmentMiddleware(mockContext, "SegmentWriteMockKey");
+        fullStorySegmentMiddleware.enableFSSessionURLInEvents = false;
+
+        IdentifyPayload identifyPayload = new IdentifyPayload.Builder()
+                .userId("userId")
+                .build();
+        when(mockChain.payload()).thenReturn(identifyPayload);
+        fullStorySegmentMiddleware.intercept(mockChain);
+
+        verifyStatic(FS.class, VerificationModeFactory.times(1));
```

Comment:
What are we verifying with these calls? Just that FS.identify is called once?

Reply:
yeah, make sure FS.identify is called once with the userid, when we intercept a Segment identify payload